src/engraving/rw/read410/tread.cpp

```diff
@@ -4155,7 +4155,7 @@ bool TRead::readProperties(Staff* s, XmlReader& e, ReadContext& ctx)
     } else if (tag == "hideSystemBarLine") {
         s->setHideSystemBarLine(e.readInt());
     } else if (tag == "mergeMatchingRests") {
-        s->setMergeMatchingRests(e.readInt());
+        s->setMergeMatchingRests(AutoOnOff(static_cast<char>(e.readInt())));
```

Comment:
```suggestion
        if (ctx.score()->mscVersion() < 450) {
            s->setMergeMatchingRests(e.readInt() ? AutoOnOff::On : AutoOnOff::Off);
        } else {
            s->setMergeMatchingRests(TConv::fromXml(e.readAsciiText(), AutoOnOff::AUTO));
        }
```
You'll also need to update `MSC_VERSION` in `constants.h`; that will need to be done for 4.5 anyway.

Reply:
I would like to push back against this. (Because I thought about it carefully.) If we are importing a pre-4.5 file, then `Sid::mergeMatchingRests` (to use your requested name) will be `false`. I think it is a much better import to import staves with the `false` setting to `AUTO`. The net effect will be that matching rests are not merged, but then if the user would like to change them, they don't have to do it one-by-one.

Reply:
Actually, I am wrong. `Sid::mergeMatchingRests` will be whatever the user has set up in their default style sheet. But I still think setting all the `false` settings to `OFF` is abusive behavior.

Reply:
I say this because the `false` setting in 4.4 is just there. I'd be willing to bet most users have only given it thought if they need to turn it on. If they've never needed to turn it on, it's just there. Therefore, those staves should be controllable by the Style setting.

Reply:
Haha, there we go already: I don't know the enum values by heart so had assumed that `Off` would be `0`, but apparently `Off` is `2`. I guess that immediately shows that it's beneficial to make the behaviour more explicit and not rely on casting between enum values and integers.

I have nothing against a variation like
```c++
        if (ctx.score()->mscVersion() < 450) {
            s->setMergeMatchingRests(e.readInt() ? AutoOnOff::ON : AutoOnOff::AUTO);
        } else {
            s->setMergeMatchingRests(TConv::fromXml(e.readAsciiText(), AutoOnOff::AUTO));
        }
```
as long as the behaviour is explicit and clear, and there is no reliance the integer values of the enum.

Reply:
Should there not be a list of current and past version number constants? I am reluctant to hardcode "450", but if I use Constant::MSC_VERSION it will break when there is a 4.6 (or anything higher than 4.5).

Reply:
It looks like there is a lot of other numerical hard-coding. Oh well. I've always tried to follow the practice that the only value that can be hard-coded is "0" (and perhaps numbers in well-known math formulae.) But I appreciate that this is an inherited code base.

Reply:
I'm planning to introduce a more sophisticated MSC_VERSION system in the near future. For now, hardcoding 450 is probably the best; I'll look for and replace those hardcoded numbers later.